packages/api/cms-api/src/dam/files/files.service.ts

```diff
@@ -240,6 +240,95 @@ export class FilesService {
         );
     }
 
+    async replace(
+        file: FileUploadInput,
+        { folderId, scope, ...assignData }: Omit<UploadFileBodyInterface, "scope"> & { scope?: DamScopeInterface },
+    ): Promise<FileInterface> {
+        let result: FileInterface | undefined = undefined;
+        try {
+            const contentHash = await this.calculateHashForFile(file.path);
+            let image: probe.ProbeResult | undefined;
+            try {
+                image = await probe(createReadStream(file.path));
+                if (image.type == "svg") image = undefined;
+                if (image !== undefined && image.orientation !== undefined && [6, 8].includes(image.orientation)) {
+                    image = {
+                        ...image,
+                        width: image.height,
+                        height: image.width,
+                    };
+                }
+            } catch (e) {
+                // empty
+            }
+
+            if (
+                image !== undefined &&
+                image.width &&
+                image.height &&
+                Math.round(((image.width * image.height) / 1000000) * 10) / 10 >= this.imgproxyConfig.maxSrcResolution
+            ) {
+                throw new CometImageResolutionException(`Maximal image resolution exceeded`);
+            }
+
+            if (folderId) {
+                const folder = await this.foldersService.findOneById(folderId);
+                if (!folder) throw new Error("Folder not found");
+                if (!this.contentScopeService.scopesAreEqual(folder.scope, scope)) {
+                    throw new Error("Folder scope doesn't match passed scope");
+                }
+            }
+
+            let exifData: Record<string, string | number | Uint8Array | number[] | Uint16Array> | undefined;
+            if (exifrSupportedMimetypes.includes(file.mimetype)) {
+                exifData = await exifr.parse(file.path);
+            }
+
```

Comment:
above code is copied from upload function

Reply:
I said that I want two separate functions, not one function with a param to switch between new upload and replace. But it's okay to extract some code. I did this in 229130615f2c376a49fd9d67d6488f8e64f50a64